cmd/antrea-agent/agent.go

```diff
@@ -234,11 +237,20 @@ func run(o *Options) error {
 	if features.DefaultFeatureGate.Enabled(features.Traceflow) {
 		go ofClient.StartPacketInHandler(stopCh)
 	}
-	// Create connection store that polls conntrack flows with a given polling interval.
+
+	// Initialize flow exporter to start go routines to poll conntrack flows and export IPFIX flow records
 	if features.DefaultFeatureGate.Enabled(features.FlowExporter) {
-		ctDumper := connections.NewConnTrackDumper(nodeConfig, serviceCIDRNet, connections.NewConnTrackInterfacer())
-		connStore := connections.NewConnectionStore(ctDumper, ifaceStore)
-		go connStore.Run(stopCh)
+		connStore := connections.NewConnectionStore(
+			connections.InitializeConnTrackDumper(nodeConfig, serviceCIDRNet, agentQuerier.GetOVSCtlClient(), o.config.OVSDatapathType),
+			ifaceStore,
+			o.pollInterval)
+		pollDone := make(chan struct{})
+		go connStore.Run(stopCh, pollDone)
+
+		flowExporter := exporter.NewFlowExporter(
+			flowrecords.NewFlowRecords(connStore),
+			o.config.FlowExportFrequency)
+		go wait.Until(func() { flowExporter.CheckAndDoExport(o.flowCollector, pollDone) }, o.pollInterval, stopCh)
```

Comment:
that's not exactly what I had in mind here. You should still have a select loop in `CheckAndDoExport`, the time duration provided here should be one second or a few seconds, and should only be used to establish the connection again in case of error.

in case of error, `CheckAndDoExport` logs the error, then returns. `wait.Until` will then take care of calling it again (which will establish the connection, and start a new for / select loop). If the connection never breaks or there is no export error, `CheckAndDoExport` will not return.

Reply:
I handled the connection creation through `if exp.process == nil` only when there is error.
I have to change this when we move the handling of connection logic to go-ipfix. Keeping this as is for now as discussed offline.